plugins/woocommerce-admin/client/products/hooks/use-product-entity-record.ts

```diff
@@ -4,9 +4,32 @@
 import { AUTO_DRAFT_NAME } from '@woocommerce/product-editor';
 import { Product } from '@woocommerce/data';
 import { useDispatch, resolveSelect } from '@wordpress/data';
-
+import { getQuery } from '@woocommerce/navigation';
 import { useEffect, useState } from '@wordpress/element';
 
+type ApiFetchOptions = Record< string, string >;
+
+type Middleware = (
+	options: ApiFetchOptions,
+	next: ( options: ApiFetchOptions ) => void
+) => void;
+
+type ApiFetch = {
+	use: ( middleware: Middleware ) => void;
+};
+
+declare const wp: { apiFetch: ApiFetch };
```

Comment:
`apiFetch` is loaded via external and the Dependency Extraction Webpack Plugin should be able to handle the transformation of the import. So would be better here to import from `@wordpress/api-fetch`?

Reply:
Good call, I tried importing from `@wordpress/data-controls` but that doesn't benefit from the Dependency Extraction Plugin in the same way it seems. Fixed in https://github.com/woocommerce/woocommerce/pull/37621/commits/7eb75469c2731b8496361cc20306d64bbf23e99b.

Reply:
Ahh, `apiFetch` exported from `@wordpress/data-controls` is something different. It's a control wrapper around `wp.apiFetch` typically used in actions or resolvers.